pkg/flow/flow_http.go

```diff
@@ -49,6 +51,36 @@ func (f *Flow) ConfigHandler() http.HandlerFunc {
 	}
 }
 
+// ComponentHandler returns an http.HandlerFunc which will delegate all requests to
+// a component named by the first path segment
+func (f *Flow) ComponentHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		vars := mux.Vars(r)
+		id := vars["id"]
+
+		// find node with ID
+		var node *controller.ComponentNode
+		for _, n := range f.loader.Components() {
+			if n.ID().String() == id {
+				node = n
+				break
+			}
+		}
+		if node == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+		handler := node.HTTPHandler()
+		if handler == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
```

Comment:
This might eventually be expensive in the future since we're asking to create a new HTTP handler for every request. Could we add some kind of TODO here noting that we might want to cache the handlers in the future?